operator/pkg/ingress/helpers.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func isCiliumDefaultIngressController(ctx context.Context, c client.Client) bool {
+	ciliumIngressClass := &networkingv1.IngressClass{}
+	if err := c.Get(ctx, types.NamespacedName{Name: ciliumIngressClassName}, ciliumIngressClass); err != nil {
+		if !errors.IsNotFound(err) {
+			log.WithError(err).Warn("Failed to load Cilium IngressClass")
+			return false
+		}
+
+		return false
+	}
+
+	isDefault, err := isIngressClassMarkedAsDefault(*ciliumIngressClass)
+	if err != nil {
+		log.WithError(err).Warn("Failed to detect default class on IngressClass cilium")
+		return false
+	}
+
+	return isDefault
+}
+
+func isCiliumManagedIngress(ctx context.Context, c client.Client, ing networkingv1.Ingress) bool {
+	ingressClassName := ingressClassName(ing)
+
+	if ingressClassName != nil && *ingressClassName == ciliumIngressClassName {
+		return true
+	}
+
+	// Check for default Ingress class
+	return (ingressClassName == nil || *ingressClassName == "") && isCiliumDefaultIngressController(ctx, c)
+}
+
+func ingressClassName(ingress networkingv1.Ingress) *string {
+	annotations := ingress.GetAnnotations()
+	if className, ok := annotations["kubernetes.io/ingress.class"]; ok {
```

Comment:
IMO we should handle this change of behavior in a separate PR.

Reply:
@meyskens : at best after migrating the actual ingress reconciler to use the controller-runtime library (& this helper function) (upcoming PR). Otherwise the logic is out of sync between secret-sync & ingress reconciliation.